### DISCUSSION TITLE
help

### DISCUSSION THREAD

**Discussion by lxm065** [sbilly/docker-zerotier-controller#7] [Q&A]
请问，我用
docker pull sbilly/zerotier-controller
docker run --rm -ti -p 4000:4000 -p 9993:9993 -p 9993:9993/udp sbilly/zerotier-controller:latest
默认配置啥都没改，启动后，web界面都能正常打开，客户端替换了PLANET，但连不上。（docker exec zerotier-one zerotier-cli join xxxxx 有显示200 join OK）
用命令zerotier-cli listpeers显示
200 listpeers a4de2130c2 - -1 - PLANET
感觉这PLANET配置不对？正常不是有ip，端口啥的吗？如：
200 listpeers 62f865ae71 50.7.252.138/9993;9555;4130 377 - PLANET

**Comment by sbilly** [comment]
正确操作 `zerotier-cli listpeers` 应该看得到 planet 地址.

需要确保：

- 参考 [Customize ZeroTierOne's controller planets](https://github.com/sbilly/docker-zerotier-controller#customize-zerotierones-controller-planets) 这里面修改并生成 planet
- 把生成的 planet 复制到 zerotier-one 终端的 `config` 路径

**Reply by lxm065** [reply]
我没修改patch/planets.json，是直接拖镜像下来启动的。是怎么修改生成呢？

后面我尝试过下载项目，解压后修改patch/planets.json，patch/planet.public and patch/planet.private
然后编译docker build --force-rm . -t sbilly/zerotier-controller:latest
在解决一堆apk报错后，现在报这些错，不知道怎么搞：
INFO: created initial world keys: previous.c25519 and current.c25519 (both initially the same)
terminate called after throwing an instance of 'int'

node/Topology.cpp:45:10: error: use of undeclared identifier 'ZT_DEFAULT_WORLD_LENGTH'
                Buffer<ZT_DEFAULT_WORLD_LENGTH> wtmp(ZT_DEFAULT_WORLD,ZT_DEFAULT_WORLD_LENGTH);
                       ^
node/Topology.cpp:46:29: error: use of undeclared identifier 'wtmp'
                defaultPlanet.deserialize(wtmp,0); // throws on error, which would indicate a bad static variable up top

**Reply by sbilly** [reply]
你可以 fork 这个项目，修改 [patch/planets.json](https://github.com/sbilly/docker-zerotier-controller/blob/main/patch/planets.json) 和 github action 里面的 [build-docker-image.yaml](https://github.com/sbilly/docker-zerotier-controller/blob/main/.github/workflows/build-docker-image.yaml) ，生成你自己的 docker image 。

**Reply by lxm065** [reply]
好的，我研究研究，3q
